catroid/src/main/java/org/catrobat/catroid/content/Look.java

```diff
@@ -377,6 +368,25 @@ public float getDirectionInUserInterfaceDimensionUnit() {
 
 	public void setRotationMode(int mode) {
 		rotationMode = mode;
+		rotateLookDatasIfNeeded(mode);
+	}
+
+	private void rotateLookDatasIfNeeded(int mode) {
+		if (mode != Look.ROTATION_STYLE_LEFT_RIGHT_ONLY && sprite.look.isFlipped()) {
+			flipAllLookDatas();
+			sprite.look.setFlipped(false);
+		}
+		boolean orientedLeft = sprite.look.getDirectionInUserInterfaceDimensionUnit() < 0;
+		if (mode == Look.ROTATION_STYLE_LEFT_RIGHT_ONLY && orientedLeft) {
+			flipAllLookDatas();
+			sprite.look.setFlipped(true);
+		}
+	}
+
+	public void flipAllLookDatas() {
+		for (LookData lookData : sprite.getLookDataList()) {
```

Comment:
The sprite is only used in [Look.java#L177](https://github.com/Catrobat/Catroid/blob/310dc99d43f993f1f58adc4d997fce554e0f0754/catroid/src/main/java/org/catrobat/catroid/content/Look.java#L177) which can be removed by applying Inversion of Control (creating a callback). This will remove the constructor parameter and member Sprite entirely.

However, I'm still not sure if we should change it... What do you think?